### PR TITLE
Reduce logging in tests

### PR DESCRIPTION
**Pull request by martint**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by mosabua** [comment]
I think we still want to proceed with this @martint

**Review by dain — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java
Comment (hashhar) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:1]: the changed logs in this class are practically very useful. cc: @mwd410 @findepi 

It helps finds cases where a catalog init is slow and cases where tests are leaking query runners or running too many of them concurrently - both introduce flakiness.

File: testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java [hunk 1]
```diff
@@ -42,15 +42,13 @@
...
 import static java.util.concurrent.TimeUnit.MINUTES;
-import static java.util.concurrent.TimeUnit.SECONDS;
 import static java.util.stream.Collectors.joining;
 
 public class LogTestDurationListener
         implements TestExecutionListener
 {
     private static final Logger log = Logger.get(LogTestDurationListener.class);
 
-    private static final Duration TEST_LOGGING_THRESHOLD = new Duration(20, SECONDS);
```
Comment (hashhar) [inline testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java:53]: why not change the threshold instead? It helps identify failures on CI when for example a test triggers GC or runs too long.

File: core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java [hunk 2]
```diff
@@ -137,7 +137,7 @@ public void loadInitialCatalogs()
                                     CatalogConnector newCatalog = catalogFactory.createCatalog(catalog);
                                     activeCatalogs.put(catalog.getCatalogHandle().getCatalogName(), newCatalog.getCatalog());
                                     allCatalogs.put(catalog.getCatalogHandle(), newCatalog);
-                                    log.info("-- Added catalog %s using connector %s --", storedCatalog.getName(), catalog.getConnectorName());
+                                    log.debug("-- Added catalog %s using connector %s --", storedCatalog.getName(), catalog.getConnectorName());
```
Comment (hashhar) [inline core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java:140]: this is useful for admins and people who help users to verify if a particular catalog was indeed loaded or not.

The prune and `createCatalog` ones are fine to change to debug though.

**Review by hashhar — COMMENTED** [review]
the PR title is very misleading - it doesn't affect JUST tests.

File: testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java (hunk 1 above)
Comment (martint) [reply testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java:53]: Junit logs the time a test took to run, so you end up with two log lines for every test

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java [hunk 3]
```diff
@@ -218,7 +218,7 @@ private DistributedQueryRunner(
         this.trinoClient = closer.register(testingTrinoClientFactory.create(coordinator, defaultSession));
 
         ensureNodesGloballyVisible();
-        log.info("Created DistributedQueryRunner in %s (unclosed instances = %s)", nanosSince(start), unclosedInstances.incrementAndGet());
+        log.debug("Created DistributedQueryRunner in %s (unclosed instances = %s)", nanosSince(start), unclosedInstances.incrementAndGet());
```
Comment (findepi) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:221]: Revert until we solve https://github.com/trinodb/trino/issues/21053

Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:221]: -> https://github.com/trinodb/trino/pull/21084

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java
Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:1]: cc @jklamer

Comment (findepi) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:1]: > * a warning log when it takes too long.

i like the idea. @jklamer would you like to followup?

File: testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java (hunk 1 above)
Comment (hashhar) [reply testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java:53]: where? I don't see it for example in https://github.com/trinodb/trino/actions/runs/8292705737/job/22697513919

Surefire logs the time per test class which is not useful enough.

File: testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java [hunk 4]
```diff
@@ -193,10 +190,7 @@ public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult
         }
 
         try {
-            Duration duration = endTest(testIdentifier);
-            if (duration.compareTo(TEST_LOGGING_THRESHOLD) > 0) {
-                log.info("Test %s took %s", testName(testIdentifier), duration);
```
Comment (findepi) [inline testing/trino-testing-services/src/main/java/io/trino/testing/services/junit/LogTestDurationListener.java:198]: A process question. This code was recently added (in https://github.com/trinodb/trino/pull/20539).
Why people involved in that PR were not asked for input or CCd before this "partial revert" PR was merged?

**Pull request merged**